src/charm.py

```diff
@@ -97,16 +101,34 @@ def _configure(self):
         self._set_alerts(container)
 
         # setup the workload (Prometheus) container and its services
-        layer = self._prometheus_layer()
         plan = container.get_plan()
-        if plan.services != layer["services"]:
-            container.add_layer("prometheus", layer, combine=True)
+        current_layer = plan.services if plan else {}
+        new_layer = self._prometheus_layer()
+
+        # If the startup arguments are the same and we use the
+        # we lifecycle, sent the config reload HTTP request instead
+
+        services_changed = DeepDiff(
```

Comment:
I still think this should be a utility method in the Pebble framework (if it isn't already) instead of every charm having to reinvent the wheel here. If there isn't such a feature in Pebble (which is not fully documented as yet) then as a stop gap measure this is fine but perhaps with a reference to an open ticket on github.

Reply:
There is no such feature in Pebble that I can see. I am also not sure it is warranted. This is the first charm I see this logic needed, and usually I go by the Rule of Three before asking for a new API.